docs/HyperIndex/Guides/rpc-requests.md

```diff
@@ -0,0 +1,324 @@
+---
+id: rpc-requests
+title: RPC Requests
+sidebar_label: RPC Requests
+slug: /rpc-requests
+---
+
+# RPC Requests guide
+
+TLDR; The repo for the code base can be found [here](todo_add_link)
+
+**Scenario**  
+In this guide, we aim to get token information for every token involved in a [Uniswap V3 pool creation event](https://docs.uniswap.org/contracts/v3/reference/core/interfaces/IUniswapV3Factory#poolcreated). For each token we should index its name, symbol and decimals.
+
+**Problem**  
+At first glance this may appear to be a simple task as we are only indexing over one event, the [pool created](https://docs.uniswap.org/contracts/v3/reference/core/interfaces/IUniswapV3Factory#poolcreated) event. However, there is one catch, have a look at the event signature: 
+
+```yaml
+PoolCreated(address indexed token0, address indexed token1, uint24 indexed fee, int24 tickSpacing, address pool)
+```
+
+The event only provides the token addresses, not the token name, symbol, and decimals. So how are we to fetch this extra information in our event handler? 
+
+**Solution**  
+In order to do this, we need to make RPC requests to the token contract.
+
+### Prerequisites
+
+This guide assumes basic familiarity with the viem library for making contract calls. Check out the [viem documentation](https://viem.sh/) for more information. This [medium article](https://medium.com/@0xape/typescript-and-viem-quickstart-for-blockchain-scripting-3f1846970b6f) is highly recommended for a gentle introduction to viem with a very similar example to what we are doing here.
+
+### Part 1: Create our Uniswap V3 `poolcreated` indexer
+
+`npx envio init`
+Contract address: `0x1F98431c8aD98523631AE4a59f267346ea31F984`
```

Comment:
At this point it's not entirely apparent which network this is. 
```suggestion
Ethereum mainnet contract address: `0x1F98431c8aD98523631AE4a59f267346ea31F984`
```
Could also simply hyperlink the address to an explorer - here's a pretty cool explorer built on top of envio
https://scope.sh/1/address/0x1F98431c8aD98523631AE4a59f267346ea31F984